src/Button/Button.component.tsx

```diff
@@ -3,113 +3,580 @@ import * as React from 'react';
+        `}
+
+    /* Outline */
+    &:focus {
+        ${OutlineStyles}
+    }
+    ${({ forceFocus, ...props }: StyledButtonProps) =>
+        forceFocus && OutlineStyles(props)}
+
+    /* Revealed State */
+    ${({ variant, revealed }: StyledButtonProps) =>
+        variant === 'primary' &&
+        revealed &&
+        css`
+            background-color: ${colors.silver.base};
+            border: solid thin ${colors.silver.base};
+            color: ${colors.charcoal.light};
+            font-weight: bold;
+        `}
+`;
+
+interface StyledHitboxProps {
+    buttonHeight: number;
+    buttonWidth?: number;
+}
+
+const StyledHitbox = styled.div<StyledHitboxProps>`
+    position: absolute;
+
+    // overlap border
+    top: -1px;
+    left: -1px;
+    right: -1px;
+    bottom: -1px;
+
+    // expand using margin to get to 3 rem tall and wide
+    margin: ${({ buttonHeight, buttonWidth }: StyledHitboxProps) =>
+        `-${(3 - buttonHeight) / 2}rem ${
+            buttonWidth && buttonWidth < 3
+                ? `-${(3 - buttonWidth) / 2}rem`
+                : '0'
+        }`};
+`;
+
+interface StyledFlipProps {
+    flipColor: string;
+}
+
+const StyledFlip = styled.div<StyledFlipProps>`
+    border-top-right-radius: ${sizes.border.radius.base};
+    border-bottom-left-radius: ${sizes.border.radius.base};
+
+    position: absolute;
+    top: -1px;
+    right: -1px;
+    width: 1.25rem;
+    height: 1.25rem;
+    content: '';
+
+    transition: ${transitionSpeed} ease opacity;
+    background: linear-gradient(
+        45deg,
+        ${({ flipColor }) => flipColor},
+        ${({ flipColor }) => flipColor} 50%,
+        ${colors.silver.base} 0
+    );
 `;
 
+const Flip = ({ colorTheme }: { colorTheme: Theme }) => (
+    <React.Fragment>
+        <StyledFlip flipColor={colorTheme.base} />
+        <StyledFlip className="flip-base" flipColor={colorTheme.light} />
+    </React.Fragment>
+);
+
 export const Button = ({
     className,
+    flip = false,
+    variant = 'primary',
+    size = 'large',
+    disabled,
+    revealed,
+    colorTheme,
+    reverse,
+    circular,
+    children,
+    icon: Icon,
     ...props
-}: ButtonProps): React.ReactElement<any> => (
-    <StyledButton
-        className={classNames('anchor-button', className)}
-        {...props}
-    />
-);
+}: ButtonProps): React.ReactElement<any> => {
+    const iconOnly = Icon && React.Children.count(children) === 0;
+
+    const iconScale = iconOnly
+        ? size === 'xsmall' ||
+          (size === 'small' && (circular || variant === 'minimal'))
+            ? 'md'
+            : 'lg'
+        : 'md';
+
+    const height = dimensions[size].height;
+    const borderRadius = circular
+        ? `${height / 2}rem`
+        : sizes.border.radius.base;
+    const width = iconOnly ? dimensions[size].height : dimensions[size].width;
+
+    // find theme from string, otherwise use defaults
+    if (typeof colorTheme === 'string') {
```

Comment:
This is the kind of polymorphism I'm usually against in JS. Not asking you to change this, but I find any kind of type-check in JS to be unreliable. In most of the cases we use this, this'll work, though. What happens if I add a random string, though? Do we want that to crash?

Reply:
Yeah I can see that. Currently if they provided a random string it'd just use the defaults.

Which reminds me, there are some props that don't go together. `flip` buttons don't have a `disabled` state, and non `flip` buttons don't have a `revealed` state. `circular` buttons should never have a `flip`. How do we want to handle situations like these? Warning? Crash?

Reply:
Warnings, but also following up with UX.